src/trace_tracker.ml

```diff
@@ -0,0 +1,87 @@
+module Op = struct
+  type t = { proc : int; variable : int; step : int }
+  let is_dependent t1 t2 = t1.proc == t2.proc || t1.variable == t2.variable
```

Comment:
Would it significantly lower the number of optimal traces to consider that two parallel `get t, get t` commute?

Reply:
Yes, potentially. Feels like there's many real-world program where that would be significant. It's quite a rabbit hole on its own, likewise we don't have to commute writes with the same value or failing cases.

Reply:
Could be explored in a follow up, this PR is already useful as it is :)

Reply:
Yeah, that sounds good. I think this part is best left as is, for it to match the implementation. Otherwise while checking new ideas it might collapse together sequences that should be explored separately under our naive dependency relation (so our algo is skipping traces, albeit immaterial ones). 

As far as I understand the lit, a dpor algorithm treats the dependency relationship as complete blackbox. We should be able to easily parametrise our exploration over whatever binary relationship as long as it conforms with Godfroid 3.1. That should make testing & adding the optims like the one proposed above more straightforward.

Reply:
For inspiration here, the TruST paper uses a nice trick where all operations are broken down into `get` and `set`, such that you would only need to consider how those two commutes (a successful CAS is a `get;set` while a failing one is just a single `get`)

Reply:
Makes sense. Do you know if they are also breaking down fetch and add, incr like that or do they do something special?

Reply:
Yes I think it's the same situation, `fad` and `exchange` always succeed and translate to a `get;set`. To be clear, they don't spell out in the paper that they do this, but their algorithm only works if you break the Atomic operations into `get` and `set` since that's the only operations they support (the translation is implicit since it's supposed to be enforced by the chosen memory model).